docs/getting-started/publish-infrastructure.md

```diff
@@ -0,0 +1,506 @@
+        kind: MySQLServerFirewallRule
+        spec:
+          forProvider:
+            serverNameSelector:
+              matchControllerRef: true
+            resourceGroupNameSelector:
+              matchControllerRef: true
+            properties:
+              startIpAddress: 0.0.0.0
+              endIpAddress: 0.0.0.0
+          reclaimPolicy: Delete
+          providerRef:
+            name: example
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-mysql"
+        - fromFieldPath: "spec.storageGB"
+          toFieldPath: "spec.forProvider.storageProfile.storageMB"
+          transforms:
+            - type: math
+              math:
+                multiply: 1024
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+```
+</div>
+<div class="tab-pane fade" id="alibaba-tab-1" markdown="1">
+```yaml
+apiVersion: apiextensions.crossplane.io/v1alpha1
+kind: Composition
+metadata:
+  name: mysqlinstances.alibaba.database.example.org
+  labels:
+    provider: alibaba
+spec:
+  writeConnectionSecretsToNamespace: crossplane-system
+  reclaimPolicy: Delete
+  from:
+    apiVersion: database.example.org/v1alpha1
+    kind: MySQLInstance
+  to:
+    - base:
+        apiVersion: database.alibaba.crossplane.io/v1alpha1
+        kind: RDSInstance
+        spec:
+          forProvider:
+            engine: mysql
+            engineVersion: "10.0"
+            dbInstanceClass: rds.mysql.s1.small
+            dbInstanceStorageInGB: 20
+            securityIPList: "0.0.0.0/0"
+            masterUsername: "myuser"
+          writeConnectionSecretsToRef:
+            namespace: crossplane-system
+          providerRef:
+            name: alibaba-provider
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+              fmt: "%s-mysql"
+        - fromFieldPath: "spec.storageGB"
+          toFieldPath: "spec.forProvider.dbInstanceStorageInGB"
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+```
+</div>
+</div>
+
+## Create Requirement
+
+Now that we have defined our new type of infrastructure (`MySQLInstance`) and
+created at least one composition that can satisfy it, we can create a
+`MySQLInstanceRequirement` in the namespace of our choosing. In each
+`Composition` we defined we added a `provider: <name-of-provider>` label. In the
+`MySQLInstanceRequirement` below we use the `compositionSelector` with
+`provider: aws` to match our AWS `Composition`. This can be adjusted to select
```

Comment:
Suggest tabs for each provider, since I just grabbed the content and ran it without reading the instructions, or maybe have the `matchLabels.provider`: REPLACE_ME in the example below.